jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,955 @@
+        new_pixels_x = []
+        new_pixels_y = []
+        for x, y in zip(badpix[0], badpix[1]):
+            pixel = (x, y)
+            if pixel not in already_found:
+                new_pixels_x.append(x)
+                new_pixels_y.append(y)
+        return (new_pixels_x, new_pixels_y)
+
+    def find_hot_dead_pixels(self, mean_image, comparison_image, hot_threshold=2., dead_threshold=0.1):
+        """Create the ratio of the slope image to a baseline slope image.
+        Pixels in the ratio image with values above ``hot_threshold`` will
+        be marked as hot, and those with ratio values less than
+        ``dead_threshold`` will be marked as dead.
+
+        Parameters
+        ----------
+        mean_image : numpy.ndarray
+            2D array containing the slope image from the new data
+
+        comparison_image : numpy.ndarray
+            2D array containing the baseline slope image to compare against
+            the new slope image.
+
+        hot_threshold : float
+            (mean_image / comparison_image) ratio value above which a pixel
+            is considered hot.
+
+        dead_threshold : float
+            mean_image / comparison_image) ratio value below which a pixel
+            is considered dead.
+
+        Returns
+        -------
+        hotpix : tuple
+            Tuple (of lists) containing x,y coordinates of newly hot pixels
+
+        deadpix : tuple
+            Tuple (of lists) containing x,y coordinates of newly dead pixels
+        """
+        # Avoid divide by zeros
+        zeros = comparison_image == 0.
+        comparison_image[zeros] = 1.
+        mean_image[zeros] += 1.
+
+        ratio = mean_image / comparison_image
+        hotpix = np.where(ratio > hot_threshold)
+        deadpix = np.where(ratio < dead_threshold)
+        return hotpix, deadpix
+
+    def get_baseline_filename(self):
+        """Query the database and return the filename of the baseline
+        (comparison) mean dark slope image to use when searching for
+        new hot/dead/noisy pixels. For this we assume that the most
+        recent baseline file for the given detector is the one to use.
+
+        Returns
+        -------
+        filename : str
+            Name of fits file containing the baseline image
+        """
+        subq = session.query(self.pixel_table.detector,
+                             func.max(self.pixel_table.entry_date).label('maxdate')
+                             ).group_by(self.pixel_table.detector).subquery('t2')
+
+        query = session.query(self.pixel_table).join(
+            subq,
+            and_(
+                self.pixel_table.detector == self.detector,
+                self.pixel_table.entry_date == subq.c.maxdate
+            )
+        )
+
+        count = query.count()
+        if not count:
+            filename = None
+        else:
+            filename = query.all()[0].baseline_file
+            print('Baseline filename: {}'.format(filename))
```

Comment:
@bhilbert4 I'd suggest switching this to a `logging` statement and moving it back to the `run` method.

Reply:
Done